user.lisp

```diff
@@ -218,10 +218,14 @@ such a case, kill the shell command to resume StumpWM."
 
 (defcommand-alias abort keyboard-quit)
 
+(defvar *quit-prompt* "Really close StumpWM? "
+  "The prompt asked before quitting StumpWM.")
 
 (defcommand quit () ()
-"Quit StumpWM."
-  (throw :top-level :quit))
+  "Quit StumpWM."
+  (when (y-or-n-p *quit-prompt*)
```

Comment:
I don't mind the yes or no prompt.  I rarely suggest doing this, but I would like the ability to configure this behavior.  When I want stumpwm to quit, ':quit' means quit, I don't want to deal with another prompt.  Could you make a global called *prompt-on-quit* and set it to nil by default and then set this to (and *prompt-on-quit* (y-or-n-p *quit-prompt*))

Reply:
Would it be better to define a special function for interactive use (along with the variable setting) so that non-interactive funcitons can call `quit` and be sure it works, but a user can call `quitsafe` (or something) to be able to have/configure the prompt?

Reply:
Well documented variables the user can tweak from their config file is more maintainable from my perspective than keeping multiple functions around that do almost the same thing.